File: ruby/lib/agentic_commerce_protocol/models/content_type.rb
# frozen_string_literal: true
# This file is @generated
module AgenticCommerceProtocol
  class ContentType
    PLAIN = "plain".freeze
    MARKDOWN = "markdown".freeze

    def self.all_vars
      @all_vars ||= [PLAIN, MARKDOWN].freeze
    end

    def initialize(value)
      unless ContentType.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #ContentType"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if ContentType.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #ContentType"
    end

    def serialize
      @value
    end
  end
end

File: java/src/main/java/com/svix/agentic_commerce_protocol/ContentType.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum ContentType implements ToQueryParam {
    PLAIN("plain"),
    MARKDOWN("markdown");
    private final String value;

    ContentType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}
